### ISSUE TITLE
Weird behavior when working with Cmds

### ISSUE THREAD

**Issue by bengobeil** [davedawkins/Sutil#23]
Hi,

In StonkWatch, branch: update-sutil

Using Sutil 1.0.0-alpha-005

I added a Cmd with AsyncMsg.

``` fsharp

type Model =
  { Portfolio: Portfolio
    CurrentPortfolioTab: PortfolioTab }

type Message =
  | SelectedPaneChanged of PortfolioTab
  | FetchPortfolio
  | PortfolioFetched of Portfolio
  
let init (): Model * Cmd<Message> =
  { Portfolio = { Positions = [] ; Balances = Undefined }
    CurrentPortfolioTab = Positions }, Cmd.ofMsg FetchPortfolio

let update (msg: Message) (model: Model): Model * Cmd<Message> =
  match msg with
  | SelectedPaneChanged portfolioTab ->
      let model =
        if portfolioTab <> model.CurrentPortfolioTab then
          { model with
              CurrentPortfolioTab = portfolioTab }
        else
          model
          
      model, Cmd.none
      
  | FetchPortfolio ->
    let msg = async {
      do! Async.Sleep 2000
      return PortfolioFetched SeedData.portfolio
    }
    model, Cmd.OfAsync.result msg
      
  | PortfolioFetched portfolio ->
    { model with Portfolio = portfolio }, Cmd.none

```

View function is a bit long, will not include.

This is the behavior I am getting this:

![2021-03-22 12-56-30](https://user-images.githubusercontent.com/64225318/112028126-3b8f9280-8b0e-11eb-8688-5a790836ad08.gif)

Basically, when I click back and forth between the tabs, the tables that get created before the 2 seconds with just stay there and not get cleaned up.

Not sure if I am doing something wrong in the code, any help appreciated!

**Comment by davedawkins** [comment]
Hi Ben
I've found the cause. The async cmd is working fine, but it's the switching between tabs that is going wrong.
The function `getViewForSelectedPane` is the argument to a call to `Bind.fragment`, and then `positionsTable` also returns a `Bind.fragment`. This *should* work, but needs a little work first. So, for now, it's best to add a containing div so that the first `Bind.fragment` can manage independently of anything the second bind is managing.

Sorry for the confusion.

Here's my fix (or workaround if you prefer):
```fs

    let getViewForSelectedPane =
      function
      | Positions ->
          let positionListStore = portfolioStore |>> (fun p -> p.Positions)

          Html.div [ /// For now, use this div to prevent Bind.fragment of Bind.fragment
            positionsTable positionListStore
          ]
          
      | Balances -> Html.text "Not done yet"
```

**Comment by bengobeil** [comment]
Ok, so from what I understand, currently, a bind.fragment with a bind.fragment passed to it doesn't work, and you recommend a div to separate the two?

**Comment by davedawkins** [comment]
For now, yes.

**Comment by bengobeil** [comment]
Great, thanks for the help! I confirm it works.

**Comment by davedawkins** [comment]
I'm going to close this issue, since it's nothing to do with Cmds, and we'll open a new issue about bindings of bindings

**Comment by davedawkins** [comment]
This is fixed in 1.0.0-beta-002
